iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -124,29 +124,55 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
                 // Force connection now.
                 // We have set the "shouldExecuteOperation" function to always try to open the connection.
                 // OpenAsync() is an idempotent call, it has the same effect if called once or multiple times on the same client.
-                await RetryOperationHelper.RetryTransientExceptionsAsync(
+                await RetryOperationHelper<Task>.RetryTransientExceptionsAsync(
                     operationName: "OpenConnection",
-                    asyncOperation: async () => await s_deviceClient.OpenAsync(cancellationToken),
+                    asyncOperation: async () => 
+                    {
+                        await s_deviceClient.OpenAsync(cancellationToken);
+                        return Task.FromResult(true);

```

Comment:
Because line 131 is awaited, you can just return `true`.

```suggestion
                        return true;
```

Using `Task.FromResult()` or `Task.Completed` only need to be used when a method signature indicates a `Task<>` or `Task` return type, but the method is not marked as `async` because it does not await any async methods.

Reply:
Based on my comment in the next file, I'm thinking we shouldn't need to return true from these methods at all.